contrib/opentimelineio_contrib/application_plugins/rv/example_otio_writer/otio_writer.py

```diff
@@ -0,0 +1,372 @@
+    )
+
+
+def _create_track(node_name):
+    """
+    Create an OTIO Track instance from an RVSequenceGroup.
+    :param node_name: `str`
+    :return: `otio.schema.Track`
+    """
+    track = otio.schema.Track(get_node_ui_name(node_name))
+    input_node_names, _ = rv.commands.nodeConnections(node_name)
+
+    # Set timing for Sequence elements
+    seq_node = group_member_of_type(node_name, "RVSequence")
+    edl = {
+        "in_frame": rv.commands.getIntProperty(seq_node + ".edl.in"),
+        "out_frame": rv.commands.getIntProperty(seq_node + ".edl.out"),
+        "cut_in_frame": rv.commands.getIntProperty(seq_node + ".edl.frame"),
+        "source": rv.commands.getIntProperty(seq_node + ".edl.source")
+    }
+
+    if edl["in_frame"] and edl["out_frame"] and edl["source"]:
+        for edl_index, source_index in enumerate(edl["source"][:-1]):
+            track.append(
+                create_otio_from_rv_node(
+                    input_node_names[source_index],
+                    in_frame=edl["in_frame"][edl_index],
+                    out_frame=edl["out_frame"][edl_index],
+                    cut_in_frame=edl["cut_in_frame"][edl_index]
+                )
+            )
+    else:
+        for input_node_name in input_node_names:
+            track.append(create_otio_from_rv_node(input_node_name))
+
+    return track
+
+
+def _create_stack(node_name, *args, **kwargs):
+    """
+    Create an OTIO Stack instance from an RVStackGroup.
+    :param node_name: `str`
+    :return: `otio.schema.Stack`
+    """
+    input_node_names, _ = rv.commands.nodeConnections(node_name)
+    stack = otio.schema.Stack(get_node_ui_name(node_name))
+    for input_node_name in input_node_names:
+        stack.append(create_otio_from_rv_node(input_node_name))
+
+    return stack
+
+
+def _create_item(node_name, in_frame=None, out_frame=None, cut_in_frame=None):
+    """
+    Create an OTIO Clip or Gap for an RVSourceGroup.
+    :param node_name: `str`
+    :param in_frame: `int`
+    :param out_frame: `int`
+    :param cut_in_frame: `int`
+    :return: `otio.schema.Clip` or `otio.schema.Gap`
+    """
+    source_node = (group_member_of_type(node_name, "RVFileSource") or
+                   group_member_of_type(node_name, "RVImageSource"))
+    source_path = rv.commands.getStringProperty(source_node + ".media.movie")[0]
+
+    # Create TimeRange
+    start_time, duration = frames_to_rational_times(
+        start_frame=in_frame or get_source_start_frame(node_name),
+        end_frame=out_frame or get_source_end_frame(node_name),
+        fps=get_source_fps(node_name)
+    )
+    if start_time or duration:
+        source_range = otio.opentime.TimeRange(
+            start_time=start_time,
+            duration=duration
+        )
+    else:
+        duration = None
+        source_range = None
+
+    # Create OTIO Gap (for movieproc) or Clip (for movies or images)
+    if source_path.endswith(".movieproc"):
+        return otio.schema.Gap(
```

Comment:
On second thought, I might actually consider these sources and not gaps.  They represent actual media.

Reply:
This was done to match the movieproc/Gap behavior in the [otio_reader.py](https://github.com/PixarAnimationStudios/OpenTimelineIO/blob/master/contrib/opentimelineio_contrib/application_plugins/rv/example_otio_reader/otio_reader.py#L230-L233)

I'm not really sure how meaningful the Clip.ExternalReference.target_url would be for a movieproc-based source (i.e. it's meaningless outside of RV), but I don't have a hard argument for or against it.